draft-ietf-ohai-ohttp.md

```diff
@@ -1065,20 +1088,45 @@ server. The use of idempotent methods might be of some use in managing replay
 risk, though it is important to recognize that different idempotent requests
 can be combined to be not idempotent.
 
-Idempotent actions with a narrow scope based on the value of a protected nonce
-could enable data submission with limited replay exposure. A nonce might be
-added as an explicit part of a request, or, if the oblivious request and target
-resources are co-located, the encapsulated keying material can be used to
-produce a nonce.
+Even without strong replay prevention, the server-chosen `response_nonce` field
+ensures that responses have unique AEAD keys and nonces even when requests are
+replayed.
+
+
+### Use of Date for Anti-Replay
 
-The server-chosen `response_nonce` field ensures that responses have unique
-AEAD keys and nonces even when requests are replayed.
+Clients SHOULD include a `Date` header field in requests.  Though HTTP requests
+often do not include a `Date` header field, the value of this field might be
+used by a server to limit the amount of requests it needs to track when
+preventing replays.
+
+A server can maintain state for requests for a small window of time over which
+it wishes to accept requests.  The server then rejects requests if the request
+is the same as one that was previously answered within that time window.
```

Comment:
If the new request has a timestamp in the new time window, it can accept it as a new value.

Generally you want a sliding window for anti-replay, so that you have a window of W = (now() - X .. now() + Y) continuously.  Then, when a request comes in with t \in W, you accept it if the enc value is unique.  You drop old values when t < now() - X. 

In practice, this isn't very efficient and you need to quantize the time window a little bit.  The anti-replay I built for TLS 0-RTT uses just two time intervals.  A single window means that there are hard discontinuities that cause rejections for very small amounts of skew when you move from one window to the next.

Either way, once you forget the old enc value, if you see the same enc value with a new value for t, it's a new request and you can accept it.